packages/adblocker/src/engine/bucket/preprocessor.ts

```diff
@@ -0,0 +1,161 @@
+import { StaticDataView } from '../../data-view';
+import Preprocessor, { Env } from '../../preprocessor';
+
+export default class PreprocessorBucket {
+  public static deserialize(view: StaticDataView, env: Env): PreprocessorBucket {
+    const preprocessors: Preprocessor[] = [];
+
+    for (let i = 0, l = view.getUint32(); i < l; i++) {
+      preprocessors.push(Preprocessor.deserialize(view));
+    }
+
+    return new this({
+      env,
+      preprocessors,
+    });
+  }
+
+  public readonly conditions: Map<string, Preprocessor>;
+  public readonly ineligible: Set<number>;
+
+  public env: Env;
```

Comment:
As described above, if we have `ineligible` we don't need to keep `env` as a reference here.

Reply:
We can keep it in the Engine instead.

Reply:
We need to keep this for later call of `PreprocessorBucket.update`.